### ISSUE TITLE
Visualiser bars full when starting OBS

### ISSUE THREAD

**Issue by orion78fr** [univrsal/spectralizer#38]
When starting OBS, I have the visualisation with the bars tall like shown in this image :

![image](https://user-images.githubusercontent.com/8122888/99099560-eb8e4580-25da-11eb-9d5c-3f8e973eee5b.png)

I don't know what's causing this, changing the audio source doesn't fix it if the other sources don't have audio.
No visible bug lines inside the logs :

```
17:59:52.465: WASAPI: Device 'Spotify Out (Virtual Audio Cable)' [48000 Hz] initialized
17:59:52.571: [spectralizer] Added audio capture to 'Spotify Audio'
```

Playing sound in the audio source (here play / pause the song in spotify) fixes it.

**Comment by Echo51** [comment]
I'm able to replicate this consistenly by creating a new one, setting it to an audio source, changing to the monstercat filter, setting filter strenght to 0, and then adjusting the detail value. Restart doesn't fix it, only removing and re-adding the visualizer, sometimes requiring a OBS restart first or it instantly breaks when you select monster cat filter

**Comment by orion78fr** [comment]
Can confirm this happen to other users too, I saw it while watching some streams.

**Comment by univrsal** [comment]
Does it always happen? I can't reproduce it here so it's a bit hard to come up with a solution

**Comment by orion78fr** [comment]
Yeah, it happen consistently on OBS start until there is sound in the source.
I'm using a virtual audio cable (from https://vac.muzychenko.net/en/download.htm) and I'm not certain but the streamer I was watching having the same bug is using a GoXLR (but may have some virtual cables too).

I may be able to setup a debugging env later this evening to look at it. Do I have to use Visual Studio or can I use mingw+CLion ?

**Comment by univrsal** [comment]
I don't think obs compiles under mingw

**Comment by orion78fr** [comment]
Damn... I will try to setup a Visual Studio env then...

**Comment by orion78fr** [comment]
Ok so I took a bit more time than I thought to setup the build env (I had to download msvc and it was heavy to download and got sidetracked during the download).
So first I can tell you this happens with monstercat filter and not the other filters. Maybe you will have the same behavior by choosing the same config as on the screenshot ?
![image](https://user-images.githubusercontent.com/8122888/110710515-4a1ca680-81fe-11eb-98ea-31dfaa666ff3.png)
Note that it doesn't happen suddenly, I have to restart OBS for this to happen

**Comment by alkaris2** [comment]
This bug happens on Linux also, what I've come to find is that if I switch it from Desktop Audio to Mic Input it works fine, but if I switch it back it just does this. But also in my findings, is that if I change my Desktop Audio output source to something else and back again, it appears to work just as normal, but this is only a temporary fix until I restart OBS.

**Comment by treierxyz** [comment]
This seems to be related to #21 as the bars are full when sound doesn't work, no matter what the properties are set to.

**Comment by iesv** [comment]
Oh, It only seems to have this problem on 1.3.4 version, but the 1.3.2 is normal.
